### DISCUSSION TITLE
Native Bluetooth Home Assistant Integration

### DISCUSSION THREAD

**Discussion by patman15** [fl4p/batmon-ha#213] [Show and tell]
Hi @fl4p!

As originally discussed in https://github.com/fl4p/batmon-ha/discussions/84#discussioncomment-8650870 I have completed a native integration to my best knowledge. Testing revealed no issues so far, but I could use some beta testers. :wink:

I also generated a plugin system to easily add further BMS types. You can find it in the [repository](https://github.com/patman15/BLE_BMS-HA/). I also wrote a short [documentation on how to add further types](https://github.com/patman15/BLE_BMS-HA/blob/main/CONTRIBUTING.md). I decided for having the plugins in the integration, as I needed some HA functionality (the BT matcher). Putting into separate API repositories, would have complicated the plugins, so I went simple. Can be changed though.

I'd love to at least share effort on the BMS querying part as I do not have other BMS types available for testing. Preferably, I'm looking forward to even join efforts or hand-over my work as I do not want to generate a competing piece of work. You already built up a great variety of supported BMS so I'd like to support that.

Let me know how to take it from here!

Thanks!

**Comment by sp0ngeboobs** [comment]
finally what I was looking for all year, a simple and working solution) my batteries are far from the home assistant and the bluetooth did not reach. I installed your addon yesterday, everything seems to be working fine. BMS Dali. I would like to see a delta between the highest and lowest charge. or charge by cells. This is important to observe so that there is no big difference between the cells. Is it possible to add it or do I have to write a whole code, because I'm not a programmer!?) Thanks!

**Reply by patman15** [reply]
@sp0ngeboobs happy to hear that you find it useful! For the requested functionality, please [raise a feature request](https://github.com/patman15/BMS_BLE-HA/issues) and then let's discuss details there. This is the wrong repo for this.

**Comment by timmills** [comment]
Hi this looks terrific - saw it reviewed on Youtube Off Grid Garage. Will the new Bluetooth stack mean that it will work with HA Bluetooth proxies? If so that would be mean we can put a BT Proxy out with batteries etc in the garage. (also means I can remotely monitor the caravan via VPN)
https://esphome.io/components/bluetooth_proxy.html

**Reply by patman15** [reply]
Credits for the add-on go to Fabian, I just did a "clone" that works with HA natively. Check the [first comment](https://github.com/fl4p/batmon-ha/discussions/213#discussion-6419834) for a link to the repository of my "clone".

**Comment by 7wells** [comment]
@fl4p 
Hi Fabian, will you be able to implement Patrick's work in your addon?

@patman15
